### PR TITLE
update CreateNewTool to execute as well

### PR DESCRIPTION
**Pull request by qcampbel**
- partly combine CreateNewTool and ExecuteSkill, since it makes more sense to do it in one step instead of two.
- renamed ExecuteSkill -->  RetryExecuteSkill. Just in case executing fails (likely due to missing arguments). May remove this later if deemed unnecessary.

### CODE REVIEW

File: mdagent/tools/maketools.py [hunk 1]
```diff
@@ -186,15 +185,21 @@ class CreateNewToolInputSchema(BaseModel):
         description="""List of all tools you have access to. Such as
         this tool, 'ExecuteSkill', 'SkillRetrieval', and maybe `Name2PDBTool`, etc."""
     )
+    execute: Optional[bool] = Field(
+        description="Whether to execute the new tool or not. default: True"
+    )
```
Comment (Jgmedina95) [inline mdagent/tools/maketools.py:190]: you can include the default value like: 
```{python}
Field(True, description="Whether to execute the new tool or not. default: True")
```
That way, you don't need to specify the value later on when calling _run

**Review by Jgmedina95 — COMMENTED** [review]
See comment in maketools file

File: mdagent/tools/maketools.py (hunk 1 above)
Comment (qcampbel) [reply mdagent/tools/maketools.py:191]: updated in latest commits

**Review by Jgmedina95 — COMMENTED** [review]
LGTM

**Review by Jgmedina95 — APPROVED** [review]

**Pull request merged**